pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+    def _validate_open_params(FilterQueryLanguage, FilterQuery,
+                              OperationTimeout, ContinueOnError):
+        """
+        Validate the fql parameters and if invalid, generate exception
+        """
+        if not FilterQueryLanguage and FilterQuery:
+            raise CIMError(
+                CIM_ERR_INVALID_PARAMETER,
+                "FilterQuery without FilterQueryLanguage definition is "
+                "invalid")
+        if FilterQueryLanguage:
+            if FilterQueryLanguage != 'DMTF:FQL':
+                raise CIMError(
+                    CIM_ERR_QUERY_LANGUAGE_NOT_SUPPORTED,
+                    _format("FilterQueryLanguage {0!A} not supported",
+                            FilterQueryLanguage))
+        ot = OperationTimeout
+        if ot:
+            if not isinstance(ot, six.integer_types) or ot < 0 \
+                    or ot > OPEN_MAX_TIMEOUT:
+                raise CIMError(
+                    CIM_ERR_INVALID_PARAMETER,
+                    _format("OperationTimeout {0!A }must be positive integer "
+                            "less than {1!A}", ot, OPEN_MAX_TIMEOUT))
+        if ContinueOnError:
+            if not isinstance(ContinueOnError, bool):
+                raise CIMError(
+                    CIM_ERR_INVALID_PARAMETER,
+                    _format("ContinueOnerror must boolean. "
+                            "Rcvd {0}.", ContinueOnError))
+
+    def _test_pull_operations_disabled(self):
+        """
+        Test if pull operations vare enabled.  If they are not, raise exception.
+        """
+        if self.disable_pull_operations:
+            raise CIMError(CIM_ERR_NOT_SUPPORTED,
+                           "Pull Operations not supported. "
+                           "disable_pull_operations=True")
+
+    ####################################################################
+    #
+    #   The pull operations
+    #   The following implement the server side of the WBEM pull
+    #   operations including the Open...(), Pull...(), and Close()
+    #
+    ####################################################################
+
+    ####################################################################
+    #
+    #   OpenEnumerateInstancePaths server request
+    #
+    ####################################################################
+
+    def OpenEnumerateInstancePaths(self, namespace, ClassName,
+                                   FilterQueryLanguage=None, FilterQuery=None,
+                                   OperationTimeout=None, ContinueOnError=None,
+                                   MaxObjectCount=None):
+        """
+        Implements WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.OpenEnumerationInstancePaths`
+        with data from the instance CIM repository.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+              (case independent).
```

Comment:
Copy & paste error from CreateInstance. Should use enumeration verbiage.

Reply:
DONE